common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -343,9 +343,23 @@ public MicrosoftStsTokenRequest createTokenRequest(@NonNull final MicrosoftStsAu
         tokenRequest.setClientId(request.getClientId());
         tokenRequest.setScope(request.getTokenScope());
         tokenRequest.setClaims(request.getClaims());
-        tokenRequest.setGrantType(TokenRequest.GrantTypes.AUTHORIZATION_CODE);
         setTokenRequestCorrelationId(tokenRequest);
 
+        // Existence of a device code inside of the response object implies Device Code Flow is being used
+        if (response.getDeviceCode() != null) {
+            tokenRequest.setGrantType(TokenRequest.GrantTypes.DEVICE_CODE);
+            tokenRequest.setDeviceCode(response.getDeviceCode());
+
+            // mTenantId from mAudience (being "common") is being added to the path in AzureActiveDirectoryAuthority
+            // Needed to omit "\common" from the endpoint so as to not break DCF
+            // mTokenEndPoint be automatically reset in future commands
+            String newTokenEndpoint = this.mTokenEndpoint.replace("/common/", "/");
```

Comment:
I get a 404 error, also happens with the authorization step if /common is placedafter the tenant

Reply:
Device Code Flow supports /common though.

Parameter | Condition | Description
tenant | Required | Can be /common, /consumers, or /organizations. It can also be the directory tenant that you want to request permission from in GUID or friendly name format.

Reply:
Tested in authorization step, supplying "common" in place of the tenant ID return "invalid_request". Even if it worked, wouldn't we run into the same issue where we would have to parse out the tenant id in order to just get /common in the endpoint?

Reply:
`Tested in authorization step, supplying "common" in place of the tenant ID return "invalid_request".`
>> What are the exact error message that you're getting? Does it have something to do with single tenant/multiple tenant applications? /common is not supported if your device is marked as a single tenant application.

` Even if it worked, wouldn't we run into the same issue where we would have to parse out the tenant id in order to just get /common in the endpoint?`
>> This is what I've been saying that this is a bug. It doesn't make sense to be to have /common append after tenantID in the first place, and we should be fixing the problem at the root cause (why /common was appended, and if it's justified.)

Reply:
Authority issue has been fixed. The error was essentially saying that there was no tenant information included.